haigha/transports/socket_transport.py

```diff
@@ -29,23 +29,33 @@ def connect(self, (host, port), klass=socket.socket):
         Connect assuming a host and port tuple.
         '''
         self._host = "%s:%s" % (host, port)
-        self._sock = klass()
+        infos = socket.getaddrinfo(host, port, 0, 0, socket.IPPROTO_TCP)
+        if not infos:
+            raise socket.error("getaddrinfo returns an empty list")
+
+        # Sort addresses such that the new IPv6 address is first.
+        for info in reversed(sorted(infos, key=lambda info: len(info[-1]))):
```

Comment:
I chose to use `reversed` because `reversed(sorted(...))` fits in 80 characters where `sorted(..., reversed=True)` does not. There's a minuscule performance hit since both `sorted` and `reversed` are eagerly evaluated, but it would only be noticeable for large numbers of resolved values.

Honestly, I think the sorting issue is moot based on the rest of your comment.